pkg/util/featuregate.go

```diff
@@ -0,0 +1,100 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/clock"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+)
+
+const (
+	DefaultCCONamespace string = "openshift-cloud-credential-operator"
+)
+
+// GetEnabledFeatureGates returns theenabled feature gates from the list provided.
+func GetEnabledFeatureGates(featureGates []configv1.FeatureGateName) (map[configv1.FeatureGateName]bool, error) {
+	featureGateAccessor, err := GetFeatureGateAccessor()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+	go featureGateAccessor.Run(ctx)
+
+	select {
+	case <-featureGateAccessor.InitialFeatureGatesObserved():
+		currentFeatureGates, _ := featureGateAccessor.CurrentFeatureGates()
+		log.Info("FeatureGates initialized", "knownFeatures", currentFeatureGates.KnownFeatures())
+	case <-time.After(1 * time.Minute):
+		log.Error(nil, "timed out waiting for FeatureGate detection")
+		return nil, fmt.Errorf("timed out waiting for FeatureGate detection")
+	}
+
+	currentFeatureGates, err := featureGateAccessor.CurrentFeatureGates()
+	if err != nil {
+		return nil, err
+	}
+
+	enabledFeatureGates := map[configv1.FeatureGateName]bool{}
+	for _, fg := range featureGates {
+		enabledFeatureGates[fg] = currentFeatureGates.Enabled(fg)
+	}
+	return enabledFeatureGates, nil
```

Comment:
Instead, let's have this function `GetCurrentFeatureGates` and return the `currentFeatureGates` from line 47. This will change the behavior in pkg/operator/controller.go as mentioned in the correlated comment.
```
return featureGateAccessor.CurrentFeatureGates()
```